script.js

```diff
@@ -1,4 +1,184 @@
+    pOneStats = Math.floor(pOneStats) + Math.floor(pOneResult);
+    pTwoStats = Math.floor(pTwoStats) + Math.floor(pTwoResult);
+    gameStats();
+    //P2 results bigger
+    // add in game mode condition
+    if (
+      (pOneResult > pTwoResult && gameMode == 1) ||
+      (pOneResult < pTwoResult && gameMode == 2)
+    ) {
+      myOutputValue =
+        myOutputValue +
+        ` but Player 1's hand is ${pOneResult}. You lost :(<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+      //P2 results smaller
+    } else if (
+      (pTwoResult > pOneResult && gameMode == 1) ||
+      (pTwoResult < pOneResult && gameMode == 2)
+    ) {
+      myOutputValue =
+        myOutputValue +
+        ` and Player 1's hand is ${pOneResult}. You won! :)<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+      //P2 results smame
+    } else if (pTwoResult == pOneResult) {
+      myOutputValue =
+        myOutputValue +
+        ` and Player 1's hand is ${pOneResult}. It's a draw! :)<br><br>The game will now restart!` +
+        leaderboard;
+      return myOutputValue;
+    }
+  } else if (turnOrOrder == -1 && input == 0) {
+    return `Welcome players, lets play a game of Beat That! <br><br>Rules:<br>Each player will take turn to roll 2 dice and choose the order they want to arrange the dice. <br><br> In normal mode, the player with the bigger hand wins while in reversed mode, the player with the smaller hand wins!<br><br> Please indicate whether you would like to play normal or reversed mode:<br>1) Normal Mode<br>2) Reversed Mode`;
+  } else if (turnOrOrder == -1 && (input == 1 || input == 2)) {
+    turnOrOrder += 1;
+    if (input == 1) {
+      gameMode = 1;
+      return `All right, lets start with the Normal game mode! Player 1 please click to roll.`;
+    } else {
+      gameMode = 2;
+      return `All right, lets start with the Reversed game mode! Player 1 please click to roll.`;
+    }
+  }
+};
+
+// updating the leader board
+var gameStats = function () {
+  if (
+    (pOneStats > pTwoStats && gameMode == 1) ||
+    (pOneStats < pTwoStats && gameMode == 2)
+  ) {
+    leaderboard = `<br><br><br> Player 1 is only leading for now, dont give up Player 2!<br><br> Player 1 Score: ${pOneStats}<br>Player 2 Score: ${pTwoStats} `;
+  } else if (
+    (pTwoStats > pOneStats && gameMode == 1) ||
+    (pTwoStats < pOneStats && gameMode == 2)
+  ) {
+    leaderboard = `<br><br><br> Player 2 is leading! I'm sure you can catch up Player 1!<br><br> Player 2 Score: ${pTwoStats} <br>Player 1 Score: ${pOneStats} `;
+  } else if (pOneStats == 0 && pTwoStats == 0) {
+    leaderboard = "";
+  } else if (pOneStats == pTwoStats) {
+    leaderboard = `<br><br><br>It's a draw!<br><br> Player 1 Score: ${pOneStats}<br>Player 2 Score: ${pTwoStats} `;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  return evaluateResults(input);
 };
```

Comment:
this `main` function into `evaluateResults` is quite redundant if all that `main` does is taking in `input` and returning the result of calling `evaluateResults` with `input` so again, consider how you can refactor `evaluateResults` into further smaller functions and call them in `main`.